### PR TITLE
Restore maximized state of the window

### PR DESCRIPTION
**Pull request by tymmesyde**
Restore the maximized state of the window at launch if it was before exiting

### CODE REVIEW

**Comment by sindresorhus** [comment]
Is it normal on Windows and Linux to restore maximized state?

File: source/index.ts [hunk 1]
```diff
@@ -327,6 +327,14 @@ function createMainWindow(): BrowserWindow {
 		config.set('lastWindowState', win.getNormalBounds());
 	});
 
+	win.on('maximize', () => {
+		config.set('maximizedState', true);
```
Comment (sindresorhus) [inline source/index.ts]:
```suggestion
		config.set('lastWindowState.isMaximized', true);
```

File: source/tray.ts [hunk 2]
```diff
@@ -19,7 +19,11 @@ export default {
 			if (win.isVisible()) {
 				win.hide();
 			} else {
-				win.show();
+				if (config.get('maximizedState')) {
+					win.maximize();
```
Comment (sindresorhus) [inline source/tray.ts]: Are you sure `.maximize()` totally replaces `.show()`?

File: source/index.ts (hunk 1 above)
Comment (sindresorhus) [reply source/index.ts]: And add a default value of false here: https://github.com/sindresorhus/caprine/blob/30d83cc3b22c279e05501ef0199d3e188885d372/source/config.ts#L86

And add it here: https://github.com/sindresorhus/caprine/blob/30d83cc3b22c279e05501ef0199d3e188885d372/source/config.ts#L14 and

Comment (tymmesyde) [reply source/index.ts]: > Is it normal on Windows and Linux to restore maximized state?

Yes, I think so.
On Windows, Spotify and other apps are doing it.

File: source/tray.ts (hunk 2 above)
Comment (tymmesyde) [reply source/tray.ts]: So I checked the doc and `maximize()` show the window but not focusing.

**Comment by sindresorhus** [comment]
I'm getting an error when Command+Q quitting the app:

Screenshot 2019-11-11 at 23 29 40

**Comment by sindresorhus** [comment]
Bump

**Comment by sindresorhus** [comment]
Bump

**Comment by sindresorhus** [comment]
Closing in favor of https://github.com/sindresorhus/caprine/issues/1268.

**Pull request closed without merging**